### ISSUE TITLE
Imported modules not transpiling correctly with es2015 preset

### ISSUE THREAD

**Issue by apoco** [babel/babel#2912]
With babel 6.1.2, using the es2015 preset, this works correctly:

```javascript
class SomeObject {
  constructor({ name }) {
    this.name = name;
  }

  greet() {
    console.log(`Hello ${this.name}`);
  }
}

const obj = new SomeObject({ name: 'World' });
obj.greet();
```

However, strange things happen when adding `import` into the mix. When running this as `file1.js`:

```javascript
import SomeObject from './file2';
const obj = new SomeObject({ name: 'World' });
obj.greet();
```

...and this as `file2.js`:

```javascript
export default class SomeObject {
  constructor({ name }) {
    this.name = name;
  }

  greet() {
    console.log(`Hello ${this.name}`);
  }
}
```

...you get the error:

> ReferenceError: name is not defined

When transpiling the class as a single-file, this is the body of the constructor:

```javascript
function SomeObject(_ref) {
    _classCallCheck(this, SomeObject);

    var name = _ref.name;

    this.name = name;
  }
```

However, when running code that imports the class, say with `babel-node` or `mocha -r babel-core/register`, this is the body of the constructor, hence the error:

```javascript
function SomeObject(_ref) {
    _classCallCheck(this, SomeObject);

    this.name = name;
  }
```

**Comment by limulus** [comment]
I’ve run into this as well. I don’t have a workaround, but I figured it is worth pointing out that #2728 is likely the same issue.

**Comment by radubrehar** [comment]
I get a simple case not working

```jsx
export default {
  name: 'test'
}
```

```jsx
import { name }  from './person'
console.log(name) //undefined
```
since it compiles to `console.log(_person.name)`, and not `console.log(person.default.name)`.

Nevertheless, ```import person from './person'`; console.log(person)` works nicely.

**Comment by jmm** [comment]
@radubrehar You're trying to do a named import of something that isn't a named export on the `person`. See #2212.

**Comment by radubrehar** [comment]
I'm using `"babel-loader": "^6.1.0"`.
@jmm thanks for the tip, looking into it

**Comment by radubrehar** [comment]
I've went over that issue, and can't seem to find a fix without modifying the code. I't quite problematic that many people have such code working in production, with babel@5, but babel@6 breaks this. Am I correct?

**Comment by jmm** [comment]
@radubrehar Yes, that scenario breaks, but it shouldn't have worked in the first place.

**Comment by radubrehar** [comment]
Just for future reference to other having the same problem.

The code above should have been
```jsx
export const name = 'test'
```
in order to work correctly.

**Comment by SteveEdson** [comment]
I have the following code https://github.com/SteveEdson/generate-assets/blob/master/src/image-assets.js, *should* this have been broken in Babel @5? Because it was working fine, but doesn't in Babel 6.

It's just being imported using `var ImageAssets = require("./image-assets");`

**Comment by jmm** [comment]
@SteveEdson That is broken by design in Babel v6, but it's different than the other issue. There's nothing inherently wrong with what you're doing or the way that worked in v5. It was an opinion on ES2015 module syntax / CJS interop. The "official" Babel v6 CJS module formatter has a different opinion that requires you to do `require("./image-assets").default`.

**Comment by SteveEdson** [comment]
@jmm I'm guessing this is covered under the 

> Remove module.exports export interop for CommonJS module formatter.

line in the changelog, however I'm not sure if it's clear enough.

I've made the change, but I'm still getting the error:

```
TypeError: ImageAssets.createIOSFolders is not a function
```

Please let me know if I should be discussing this problem elsewhere, if it is not related to this issue. Thanks.

**Comment by SteveEdson** [comment]
Ignore that, I'd forgotten to change the import in my tests. This appears to be fixed now, although I'm not sure that I like this new behaviour.

Thanks

**Comment by jmm** [comment]
@SteveEdson Yes, that's the relevant line. I haven't actually looked yet to see how it's documented -- if that's all there is I agree it's not enough. I think there are a bunch of other docs, like a migration guide, probably in the works.

I'm not sure off the top of my head what the issue you're experiencing now is. I think there have been some issues with default exporting classes in general.

> Please let me know if I should be discussing this problem elsewhere, if it is not related to this issue. Thanks.

The best place at the moment would be slack (see the README). If you're feeling adventurous you could try to help get the ball rolling at https://discuss.babeljs.io/.
